### ISSUE TITLE
Federation entity resolver generation

### ISSUE THREAD

**Issue by nhibberd** [99designs/gqlgen#1454]
A combination of https://github.com/99designs/gqlgen/issues/1388 & https://github.com/99designs/gqlgen/issues/1429

### What happened?
`{Entity}Resolver`'s are not generated (example [example/federation/reviews/graph/schema.resolvers.go](https://github.com/99designs/gqlgen/blob/master/example/federation/reviews/graph/schema.resolvers.go#L13-L44)) without specifying an explicit custom model (e.g. [example/federation/reviews/graph/model/models.go](https://github.com/99designs/gqlgen/blob/master/example/federation/reviews/graph/model/models.go))

Replication in the examples:

```
cd example/federation/reviews
rm graph/model/models.go
go generate server.go
```

```
validation failed: packages.Load: gqlgen/example/federation/reviews/graph/schema.resolvers.go:38:40: ProductResolver not declared by package generated
gqlgen/example/federation/reviews/graph/schema.resolvers.go:41:37: UserResolver not declared by package generated

exit status 3
```

### What did you expect?

I expected that generated Entitys would also generate the EntityResolver's? Is this not the case by design?

### Minimal graphql.schema and models to reproduce

```
type Review {
    body: String!
}

extend type User @key(fields: "id") {
    id: ID! @external
    reviews: [Review]
}
```

### versions
 - `gqlgen version`? v0.13.0
 - `go version`? 1.14
 - dep or go modules?

**Comment by nhibberd** [comment]
Workaround for anyone else who comes across this:


Add a custom model for the type for the entity being extended, in this example `User`
```golang
package model

type User struct {
	ID string `json:"id"`
}

func (User) IsEntity() {}
```

And the model to the config - `autobind` 

```
autobind: []
  - "mymodel/model"
```

**Comment by MichaelJCompton** [comment]
Yep, looks like this doesn't quite work correctly without a custom model.  

The work around (use custom models) is annoying, but looks like it generates the correct code and isn't functionally different.

**Comment by MichaelJCompton** [comment]
Actually, looks like it's specifically to do with the fields generated in the model

```graphql
extend type Product @key(fields: "upc") {
    upc: String! @external
    reviews: [Review]
}
```

Won't work with a model like

```go
type Product struct {
	Upc        string    `json:"upc"`
	Reviews []*Review `json:"reviews"`
}
```

The `Review` field means that the `ProductResolver` doesn't get generated.
